### PR TITLE
CI: Migrate from Intel to Apple Silicon

### PR DESCRIPTION
**Pull request by spencertransier**
## Description
This PR migrates WCiOS from the Intel CI stack over to our new Apple Silicon CI by upgrading to the Xcode 15.1 image, which is only available on Apple Silicon.

- It removes the `cache-builder.yml` pipeline because that isn't needed anymore with the changes in the Apple Silicon infrastructure tooling. 
- It removes the `git-s3-cache` plugin from the pipelines because that also isn't needed with the AS infrastructure
- Uses the correct whitespace character so that tests can pass (similar to https://github.com/woocommerce/woocommerce-ios/pull/11352/commits/c2a55f8585460fc991e7b97451af3b1608d01307)
-  Merging into `feat/xcode-15-ci` because @jaclync has been working on other Xcode 15-related fixes in that branch
- It's expected that the CI status checks for this PR will fail because those are still pointing to the Intel-based pipeline

Corresponding Terraform PR: https://github.com/Automattic/buildkite-ci/pull/339

## Testing instructions
It looks like there is a problem with the `test_load_support_screen` UI test: https://buildkite.com/automattic/woocommerce-ios-apple-silicon/builds/5

That looks similar to the other UI testing issues with Xcode 15 that @jaclync and @pachlava were discussing in https://github.com/woocommerce/woocommerce-ios/pull/11352, so I suggest we go ahead and merge this PR into that branch so they can continue working on a solution for all of the UI testing issues since those are an Xcode 15 problem and not an Apple Silicon CI problem. 

The "Build" and "Unit Tests" steps did complete successfully on the Apple Silicon CI 🎉 . You'll need to check the `WooCommerce iOS - Apple Silicon` pipeline manually because the regular `WooCommerce iOS` pipeline is pointing to the Intel nodes. https://buildkite.com/automattic/woocommerce-ios-apple-silicon/builds/

If more changes are made to this branch, you can go to that link and click "New Build". Under branch, type `move/wcios-to-apple-silicon-ci` and click "Create Build". That will kick off a build for this PR branch. 

## After Merging
- [ ] Update the description for https://github.com/woocommerce/woocommerce-ios/pull/11352 by adding these steps: 

```
When merging:
- [ ] Merge the corresponding Terraform PR to apply the configuration to the Buildkite pipeline: https://github.com/Automattic/buildkite-ci/pull/339

After merging:
- [ ] @spencertransier will post a P2 informing the WCiOS developers that they'll need to rebase on trunk in order for CI checks to pass on PRs 
```

---
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by AliSoftware — APPROVED** [review]

**Comment by AliSoftware** [comment]
@spencertransier https://github.com/Automattic/buildkite-ci/pull/339 has been approved, so once you deploy it, feel free to re-trigger a new build for this PR on the main `woocommerce-ios` pipeline to validate that the main pipeline will now pass and use Apple Silicon infra.

That being said, I'm guessing that you won't deploy https://github.com/Automattic/buildkite-ci/pull/339 until `feat/xcode-15-ci` is ready to be merged and land in the main branch, otherwise switching to the cluster will make all other PRs fail 🙃  So I'm not sure how you planned to handle a smooth transition.

I still approved the PR to unblock, but another approach to this that would make the transition easier would have been to:

 - First make a PR to just change use `git-s3-cache#auto-disable-on-arm64` plugin version (which still runs the plugin on Intel but skip it on ARM)
 - Merge that PR in `feat/xcode-15-ci`, then land `feat/xcode-15-ci` into `trunk`.
    - At that point, WCiOS should be able to run on both infras.
    - Which also means that open PRs will have time to merge `trunk` back into their branch and all without being blocked, and that switching to ARM won't break open PRs (while if you switch earlier and there are still PRs using Xcode 14 and don't have the fixes from `feat/xcode-15-ci` yet, they would break, right?)
 - Once everyone is building with Xcode 15 (and builds are passing in _both_ Intel and ARM infras), then make the switch to the cluster (`buildkite-ci` repo)
 - Once this is confirmed to work and all PRs are still going green while being now build in ARM infra, clean things up by removing `git-s3-cache` and the `cache-builder.yml` like you did in this PR, and getting rid of `woocommerce-ios-apple-silicon` pipeline.

**Comment by spencertransier** [comment]
> At that point, WCiOS should be able to run on both infras.

@AliSoftware The problem is that we don't have an Xcode 15.1 Intel image, only an Apple Silicon one. So this couldn't run on both infras anyway due to that. It would fail while trying to find the non-existent Intel 15.1 image.

**Comment by AliSoftware** [comment]
> The problem is that we don't have an Xcode 15.1 Intel image, only an Apple Silicon one

Ah, good point 👍

**Pull request merged**
